github/ci/jenkins-master/tasks/main.yml

```diff
@@ -0,0 +1,82 @@
+---
+- name: Install jenkins RPM repo
+  get_url: url=http://pkg.jenkins-ci.org/redhat/jenkins.repo dest=/etc/yum.repos.d/jenkins.repo
+  retries: 5
+  delay: 10
+- name: Install jenkins GPG key
+  rpm_key:
+    state: present
+    key: http://pkg.jenkins-ci.org/redhat/jenkins-ci.org.key
+- name: Install jenkins
+  dnf:
+    name: '{{ item }}'
+    state: latest
+  with_items:
+    - java
+    - jenkins  
+    - firewalld  
+- name: Enable Firewalld
+  systemd:
+    name: firewalld
+    enabled: yes
+- name: Enable Jenkins
+  systemd:
+    name: jenkins
+    enabled: yes
+- lineinfile:
+    path: /etc/sysconfig/jenkins
+    regexp: '^JENKINS_JAVA_OPTIONS='
+    line: 'JENKINS_JAVA_OPTIONS="-Djava.awt.headless=true -Djenkins.install.runSetupWizard=false"'    
+- name: Ensures init.groovy.d directory exists
+  file: 
+    path: /var/lib/jenkins/init.groovy.d/ 
+    state: directory    
+- template:
+    src: "{{ role_path }}/templates/01-plugins.groovy"
+    dest: /var/lib/jenkins/init.groovy.d/01-plugins.groovy
+- template:
+    src: "{{ role_path }}/templates/02-security.groovy"
+    dest: /var/lib/jenkins/init.groovy.d/02-security.groovy
+- template:
+    src: "{{ role_path }}/templates/03-seed.groovy"
+    dest: /var/lib/jenkins/init.groovy.d/03-seed.groovy
+- template:
+    src: "{{ role_path }}/templates/04-jobs.groovy"
+    dest: /var/lib/jenkins/jobs.groovy
+- template:
+    src: "{{ role_path }}/templates/05-github.groovy"
+    dest: /var/lib/jenkins/init.groovy.d/05-github.groovy
+- template:
+    src: "{{ role_path }}/templates/06-done.groovy"
+    dest: /var/lib/jenkins/init.groovy.d/06-done.groovy
+- name: Make sure the Jenkins service is running
+  systemd:
+    state: restarted
+    daemon_reload: yes
+    name: jenkins
+- name: Wait for Jenkins to be fully provisioned
+  wait_for:
+    path: /var/lib/jenkins/done
+    timeout: 300
+- name: Delete the file which indicates a completed jenkins provision
+  file:
+    state: absent
+    path: "/var/lib/jenkins/done"
+- firewalld:
+    zone: public
+    port: 8080/tcp
+    permanent: true
+    state: enabled
+- firewalld:
+    zone: public
+    service: http
+    permanent: true
+    state: enabled
+- name: Delete startup scripts with sensible data after jenkins started
```

Comment:
```
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and 
make sure become_method is 'sudo' (default).
This feature will be removed in a 
future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
```

Sounds like a future bug report waiting to happen...